packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -163,33 +165,60 @@ export class PipelineEditor extends React.Component<
   widgetContext: DocumentRegistry.Context;
   position = 10;
   node: React.RefObject<HTMLDivElement>;
+  properties: any;
   propertiesInfo: any;
 
   constructor(props: any) {
     super(props);
     this.app = props.app;
     this.browserFactory = props.browserFactory;
-    this.canvasController = new CanvasController();
-    this.canvasController.setPipelineFlowPalette(palette);
     this.widgetContext = props.widgetContext;
-    this.widgetContext.ready.then(() => {
-      this.canvasController.setPipelineFlow(this.widgetContext.model.toJSON());
-    });
+
     this.toolbarMenuActionHandler = this.toolbarMenuActionHandler.bind(this);
     this.contextMenuHandler = this.contextMenuHandler.bind(this);
     this.contextMenuActionHandler = this.contextMenuActionHandler.bind(this);
     this.clickActionHandler = this.clickActionHandler.bind(this);
     this.editActionHandler = this.editActionHandler.bind(this);
     this.tipHandler = this.tipHandler.bind(this);
 
-    this.state = { showPropertiesDialog: false, propertiesInfo: {} };
+    // Read config from the backend
+    const palette: any = null;
+
+    Promise.resolve([
+      RequestHandler.makeGetRequest('elyra/pipeline/config/palette', false)
+    ])
+      .then(([palette]) => {
+        console.log(palette);
+      })
+      .catch((reason: Error) => {
+        console.error(reason.message);
+      });
```

Comment:
```suggestion
    const palette = await RequestHandler.makeGetRequest('elyra/pipeline/config/palette', false);
```

Reply:
note that the key here is `await`, this tells the code to wait for the Promise to resolve then set `palette` to its result.